src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11170,7 +11251,8 @@ private ParameterSyntax ParseLambdaParameter()
             }
 
             SyntaxToken paramName = this.ParseIdentifierToken();
-            var parameter = _syntaxFactory.Parameter(default(SyntaxList<AttributeListSyntax>), modifiers.ToList(), paramType, paramName, null);
+            var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
+            var parameter = _syntaxFactory.Parameter(default(SyntaxList<AttributeListSyntax>), modifiers.ToList(), paramType, paramName, exclamation, null);
```

Comment:
note: `!` is  new lanauge feature, so it likely needs to be behind a feature flag.  i.e. this should report an error prior to whatever version of C# it is that you're parsing.

Reply:
I suggest adding a prototype comment about this and addressing in a future PR